### PR TITLE
Deprecate displayed coordinates

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR deprecates `layer.displayed_coordinates` and `layer.expand_shape` which will help us in getting the dims information off the layer and make #2249 much easier to review.

It also moves the shapes `layer._move` method into the mouse_bindings file where it probably belongs.

I'm pretty happy with how this turned out. There's still lots more to do for example every `layer._dims_displayed` should ultimately disappear too, but than can come with time.

I think this is ready for review.

## Type of change

- [ ] Deprecation

### CODE REVIEW

File: napari/layers/shapes/shapes.py [hunk 1]
```diff
@@ -2141,6 +2134,14 @@ def expand_shape(self, data):
         data_full : array
             Full D dimensional data array of the shape.
         """
+        warnings.warn(
```
Comment (Carreau) [inline napari/layers/shapes/shapes.py:2142]: do you want to deprecated in the docstring as well, so that it's shown in the documentation ?

**Review by Carreau — APPROVED** [review]

File: napari/layers/base/base.py [hunk 2]
```diff
@@ -1020,6 +1020,13 @@ def _update_draw(self, scale_factor, corner_pixels, shape_threshold):
     @property
     def displayed_coordinates(self):
         """list: List of currently displayed coordinates."""
+        warnings.warn(
+            "displayed_coordinates is deprecated and will be removed in version 0.4.8."
+            " It should no longer be used as layers should will soon not know"
+            " which dimensions are displayed.",
```
Comment (jni) [inline napari/layers/base/base.py]: @sofroniewn in the interim, is there a way for users to discover this information in a more future-proof way?

File: napari/layers/shapes/_shapes_mouse_bindings.py [hunk 3]
```diff
@@ -224,20 +236,17 @@ def vertex_insert(layer, event):
...
+    vertices = layer._data_view.shapes[index].data
     if not closed:
         if int(ind) == 1 and loc < 0:
             ind = 0
         elif int(ind) == len(vertices) - 1 and loc > 1:
             ind = ind + 1
 
     # Insert new vertex at appropriate place in vertices of target shape
-    vertices = np.insert(vertices, ind, [layer.displayed_coordinates], axis=0)
+    vertices = np.insert(vertices, ind, [layer.coordinates], axis=0)
```
Comment (jni) [inline napari/layers/shapes/_shapes_mouse_bindings.py:247]: I don't understand why this change works at all?

File: napari/layers/shapes/shapes.py [hunk 4]
```diff
@@ -2141,6 +2134,14 @@ def expand_shape(self, data):
...
+            "expand_shape is deprecated and will be removed in version 0.4.8."
+            " It should no longer be used as layers should will soon not know"
+            " which dimensions are displayed.",
```
Comment (jni) [inline napari/layers/shapes/shapes.py]: As before, I don't love removing functionality without a stopgap for how to do it in the future... Do you have an idea of how we will do this? Perhaps rather than deprecating it altogether, add a required argument of `to_dims` or similar?

**Review by jni — COMMENTED** [review]
@sofroniewn I've left some potentially minor, potentially major :grimacing:  comments! :sweat_smile: I agree that these things need to be deprecated but I dislike that we don't offer any alternative code paths to people who want to update their code in light of the deprecation.

File: napari/layers/base/base.py (hunk 2 above)
Comment (sofroniewn) [reply napari/layers/base/base.py]: Yes, I've updated that

File: napari/layers/shapes/_shapes_mouse_bindings.py (hunk 3 above)
Comment (sofroniewn) [reply napari/layers/shapes/_shapes_mouse_bindings.py:247]: It's only possible because we also changed the line above `vertices = layer._data_view.shapes[index].data`. Basically we never drop down to the 2D representation, so we can always stay in the nD representation

File: napari/layers/shapes/shapes.py (hunk 4 above)
Comment (sofroniewn) [reply napari/layers/shapes/shapes.py]: This really is an internal method that I don't expect a user to have ever used, and now we don't need it. It should never have been public, but I've also added some info about trying to stay using the full nD representation the whole time. See also my comment here https://github.com/napari/napari/pull/2327/files#r586552820

File: napari/layers/shapes/shapes.py (hunk 1 above)
Comment (sofroniewn) [reply napari/layers/shapes/shapes.py:2142]: Done, can you let me know if I've done the formatting right. Thanks!!

Comment (sofroniewn) [reply napari/layers/shapes/shapes.py:2142]: (both here and above in `displayed_coordinates`)

File: napari/layers/base/base.py (hunk 2 above)
Comment (sofroniewn) [reply napari/layers/base/base.py]: Actually @jni this is a little more complicated as we'll deprecate `layer.coordinates` too, probably in this release too, as we don't want to store cursor information on the layer, see #2249. My recommendation is we merge with this warning now and then update before the next release?

**Comment by sofroniewn** [comment]
> @sofroniewn I've left some potentially minor, potentially major 😬 comments! 😅 I agree that these things need to be deprecated but I dislike that we don't offer any alternative code paths to people who want to update their code in light of the deprecation.

Points well taken, I've added the alternative code path for `layer.displayed_coordinates` but we might need to update it again before the next release, see detailed comment above

**Review by jni — APPROVED** [review]
